addons/mail/static/src/components/composer/composer.scss

```diff
@@ -13,258 +14,61 @@
 }
 
 // ------------------------------------------------------------------
 // Style
 // ------------------------------------------------------------------
 
-// TODO FIXME o-open on the button should be enough.
-// Style of button when popover is "open" comes from web.Popover, and we can't
-// define a modifier on .o_Composer_button due to not being aware of Popover's
-// state in context of template. https://github.com/odoo/owl/issues/693
-.o_is_open .o_Composer_toolButton {
-    background-color: gray('200');
-}
-
-.o_Composer {
-    &:not(.o-messaging-in-editing) {
-        background-color: gray('100');
-    }
-
-    &.o-is-in-thread-view {
-        background-color: $white;
-    }
-}
-
-.o_Composer_actionButton.o-last.o-has-current-partner-avatar.o-composer-is-compact {
-    border-radius: 0 $o-mail-rounded-rectangle-border-radius-lg $o-mail-rounded-rectangle-border-radius-lg 0;
-}
-
-.o_Composer_button.o-composer-is-compact {
-    border-left: none; // overrides bootstrap button style
-
-    :last-child {
-        border-radius: 0 3px 3px 0;
-    }
-}
-
 .o_Composer_buttonDiscard {
-    border: $border-width solid lighten(gray('400'), 5%);
-}
-
-.o_Composer_buttons {
-    border: 0;
-
-    &.o-messaging-in-editing {
-        border-right: $border-width solid $border-color;
-    }
-}
-
-.o_Composer_coreMain:not(.o-composer-is-compact) {
-    background: $white;
-    border: $border-width solid $border-color;
-    border-radius: $o-mail-rounded-rectangle-border-radius-lg;
-}
-
-.o_Composer_currentPartnerAvatar {
-    object-fit: cover;
-}
-
-.o_Composer_textInput {
-    appearance: none;
-    outline: none;
-    background-color: $white;
-    border: $border-width solid $border-color;
-    border-right: 0;
-
-    &:not(.o-composer-is-compact) {
-        border: 0;
-        border-radius: $o-mail-rounded-rectangle-border-radius-lg;
-    }
-
-    &.o-has-current-partner-avatar.o-composer-is-compact {
-        border-radius: $o-mail-rounded-rectangle-border-radius-lg 0 0 $o-mail-rounded-rectangle-border-radius-lg;
-    }
-}
-
-.o_Composer_toolButton {
-    border: 0; // overrides bootstrap btn
-    background-color: $white; // overrides bootstrap btn-light
-    color: gray('600'); // overrides bootstrap btn-light
-    border-radius: 50%;
-
-    &.o-open {
-        background-color: gray('200');
-    }
-}
-
-.o_Composer_toolButtons {
-    background-color: $white;
-    border-top: $border-width solid $border-color;
-    border-bottom: $border-width solid $border-color;
-
-    &:not(.o-composer-is-compact) {
-        border-bottom: 0;
-        border-radius: initial;
-    }
-
-    &:last-child:not(.o-composer-has-current-partner-avatar) {
-        border-right: $border-width solid lighten(gray('400'), 5%);
-    }
+    border: $border-width solid lighten($gray-400, 5%);
 }
 
-.o_Composer_toolButtonSeparator {
-    border-left: $border-width solid $border-color;
+.o_Composer_toolButtons:last-child:not(.o-composer-has-current-partner-avatar) {
```

Comment:
Indeed no cases can have this style rule, so dead code.
I think the reasoning of this LOC, when introduced in the past, was to have right border in chat window.
<img width="339" alt="Screenshot 2022-05-16 at 15 36 10" src="https://user-images.githubusercontent.com/6569390/168605258-0601bd27-9daf-4f41-b7c3-8b186a3b1e4e.png">
But actually it would look off if there was a right border.

There's actually a left border is that is slightly off, but it's not a big deal because it's partially hidden by the shadowing on the left side of chat window.